currency/pairs.go

```diff
@@ -127,11 +132,11 @@ func (p Pairs) Contains(check Pair, exact bool) bool {
 // and removes it from the list of pairs
 func (p Pairs) RemovePairsByFilter(filter Code) Pairs {
```

Comment:
can remove the element from the array as soon as its found rather than ranging over the whole thing n appending 1 by 1